OPHD/UI/Core/RadioButtonGroup.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include "TextControl.h"
+#include "Label.h"
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/Signal/Signal.h>
+#include <NAS2D/Signal/Delegate.h>
+#include <NAS2D/EventHandler.h>
+#include <NAS2D/Resource/Image.h>
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+#include <NAS2D/MathUtils.h>
+
+#include <algorithm>
+#include <string>
+
+class RadioButtonGroup : public Control
+{
+private:
+    class RadioButton : public TextControl
+    {
+    public:
+        RadioButton(RadioButtonGroup* parentContainer, std::string newText, NAS2D::DelegateX<void> delegate);
+        ~RadioButton() override;
+        void checked(bool toggle);
+        bool checked() const;
+        void text(const std::string& text);
+        const std::string& text() const;
+        void update() override;
+
+    protected:
+        void onResize() override;
+        void onTextChange() override;
+        void onMouseDown(NAS2D::EventHandler::MouseButton button, int x, int y);
+
+    private:
+        const NAS2D::Font& mFont;
+        const NAS2D::Image& mSkin;
+        Label mLabel;
+        RadioButtonGroup* mParentContainer{nullptr};
+        bool mChecked{false};
+        NAS2D::DelegateX<void> mSignal;
+    };
+
+public:
+	struct ButtonInfo
+	{
+		std::string name;
+		NAS2D::DelegateX<void> delegate;
+	};
+
+	RadioButtonGroup() = default;
+	RadioButtonGroup(std::vector<ButtonInfo> buttonInfos);
+	void onMove(NAS2D::Vector<int> displacement) override;
+	void update() override;
+	void clear();
+	void select(std::size_t index);
+	void select(RadioButtonGroup::RadioButton* button);
```

Comment:
This section probably needs to be broken up a bit with blank lines. Special member functions (constructors, destructors, copy/move operators, etc.) are usually separated from custom named members by a blank line. If member functions are grouped into related sets (like `clear`, and the two overloads of `select`), there may also be a blank line between them and other sets of methods.

The `onMove` event handler should probably be in a `protected` section. It's not expected that client code can call this method directly.

Reply:
Done, I also re-arranged the .cpp files structure to better match the new order in the .h file.
The `onMove` has been moved to the `protected` section.

Reply:
Ahh excellent. Always good to have the .cpp and .h file list things in the same order.